File: backend/backend/Models/Book.cs
namespace backend.Models;

public class Book
{
    public Guid Id { get; set; }
    
    public string Title { get; set; }
    
    public string Summary { get; set; }
    
    public string Isbn { get; set; }
    
    public string Language { get; set; }
    
    public DateTime PublishedAt { get; set; }
    
    public Guid AuthorId { get; set; }
    
    public Author Author { get; set; }
}

File: frontend/src/entities/Book.ts
import type { Guid } from "guid-typescript";
import type { ReservationState } from "./enums/ReservationState";

interface BookReservationDto {
  id: Guid;
  state: ReservationState;
}

interface BookCopyDto {
  id: Guid;
  code: string;
  isTaken: boolean;
}

export type Book = {
  id: Guid;
  title: string;
  summary: string;
  isbn: string;
  language: string;
  publishedAt: Date;
  author: string;
  printingHouse: string;
  genres: string[];
  publisher: string;
  activeReservation: BookReservationDto | null;
  copies: BookCopyDto[];
  freeCopyCount: number;
};
